libraries/AP_HAL_Linux/RCInput.h

```diff
@@ -47,7 +47,8 @@ class RCInput : public AP_HAL::RCInput {
     void _process_rc_pulse(uint16_t width_s0, uint16_t width_s1);
     void _update_periods(uint16_t *periods, uint8_t len);
 
-    volatile bool new_rc_input;
+    volatile uint32_t rc_input_count;
```

Comment:
I don't think this use of volatile is correct.... it doesn't protect against one processor seeing the update to rc_input_count before other fields (e.g.  _num_channels). We should be using an atomic type here to guarantee there's a write fence before. See the atomic usage in RingBuffer.h.